### PR TITLE
fix(Authoring): Check asset file is in asset folder

### LINKED ISSUE #188: Check that project asset file is actually in project asset folder
When a teacher makes a request to download or delete an asset file, make sure the file is actually in the asset folder and not referencing something outside of the asset folder. The teacher can perform a path traversal exploit by specifying something outside of the asset folder by providing a file name that looks something like "../../../password.txt".

The download asset endpoint is this (where 2 is the unit number)
http://localhost:81/api/author/project/asset/2/download?assetFileName=../password.txt

The delete asset endpoint is this (where 2 is the unit number and assetFileName is provided in the form-data)
http://localhost:81/api/author/project/asset/2/delete

Also make sure there are no potential path traversal exploits when importing a step since importing a step also copies assets from one folder to another.

### PR DESCRIPTION
**Pull request by geoffreykwan**
## Changes

Check that asset file is actually referencing a file in the asset folder when
- Downloading an asset file
- Deleting an asset file
- Importing a step that references asset files

## Test

Log in as a teacher to perform these tests.

- Try downloading an asset file that is not in the unit asset folder. Change the 2 to your unit number and change the value of assetFileName to a file path that's on your computer relative to the unit's asset folder like "../../../../../../Dockerfile". You used to be able to download any file but now you shouldn't be able to.
```
http://localhost:81/api/author/project/asset/2/download?assetFileName=../password.txt
```

- Try deleting an asset file that is not in the unit asset folder. I used Postman to test this. Use the GET method and the url below (change the 2 to your unit number). Set the SESSION cookie variable copied from your signed in teacher session in the browser. Set the Body to form-data and set key to assetFileName and the value to something on your computer (I created a file in my src/main/webapp/curriculum/2 folder named password.txt so the value would be ../password.txt). You used to be able to delete any file but now you shouldn't be able to.
```
http://localhost:81/api/author/project/asset/2/delete
```

- Try importing a step that contains references to asset files. This is the JSON component content for an HTML component in the step I imported. Create my-image.jpg and model/html-file.html in your unit asset folder. Create password.txt as a sibling to the asset folder (so that it is not in the asset folder). When you import the step, it should not copy the password.txt but it should copy the my-image.jpg and the model/html-file.html.
```
{
    "id": "6u1x8l2g1u",
    "type": "HTML",
    "prompt": "",
    "showSaveButton": false,
    "showSubmitButton": false,
    "html": "<p>Enter html here</p>\n<p><img src=\"../password.txt\" /></p>\n<p><img src=\"my-image.jpg\" /></p>\n<p><img src=\"model/html-file.html\" /></p>"
}
```

Closes #188

### CODE REVIEW

File: src/test/java/org/wise/vle/utils/FileManagerTest.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    assertIsFilePathInFolder("/src/main/webapp/curriculum/15/assets", "my-image.jpg", true);
+  }
+
+  @Test
+  public void isFilePathInFolder_FilePathIsInChildFolder_ShouldReturnTrue() {
+    assertIsFilePathInFolder("/src/main/webapp/curriculum/15/assets", "model/my-image.jpg", true);
+  }
+
+  @Test
+  public void isFilePathInFolder_FilePathIsNotInFolder_ShouldReturnTrue() {
```
Comment (hirokiterashima) [inline src/test/java/org/wise/vle/utils/FileManagerTest.java]:
```suggestion
  public void isFilePathInFolder_FilePathIsNotInFolder_ShouldReturnFalse() {
```

File: src/main/java/org/wise/vle/utils/FileManager.java [hunk 2]
```diff
@@ -85,21 +85,24 @@ public static void copy(File src, File dest) throws IOException {
...
    *                                  when file name references in the content need to be changed
    *                                  due to file name conflicts.
-   * @param fromProjectAssetsFolder
-   *                                  the asset folder in the from project
-   * @param toProjectAssetsFolder
-   *                                  the asset folder in the to project
+   * @param fromProjectAssetsFolderPath
+   *                                  the asset folder path in the from project
+   * @param toProjectAssetsFolderPath
+   *                                  the asset folder path in the to project
```
Comment (hirokiterashima) [inline src/main/java/org/wise/vle/utils/FileManager.java]:
```suggestion
   * @param fromProjectAssetsFolderPath the asset folder path in the from project
   * @param toProjectAssetsFolderPath the asset folder path in the to project
```

**Review by hirokiterashima — APPROVED** [review]
Looks good. Added minor change suggestions.

**Review by breity — APPROVED** [review]
LGTM.

**Pull request merged**